src/util/amount.rs

```diff
@@ -1375,4 +1449,47 @@ mod tests {
         let value_without: serde_json::Value = serde_json::from_str("{}").unwrap();
         assert_eq!(without, serde_json::from_value(value_without).unwrap());
     }
+
+    #[cfg(feature = "serde")]
+    #[test]
+    fn serde_as_sat_opt() {
+        use serde_json;
```

Comment:
There is [`serde_test`](https://docs.rs/serde_test/1.0.119/serde_test/) in case you don't want to rely on JSON-specific semantics when testing your implementations.

Reply:
`serde_test` for example makes it quite easy to test `Visitor` implementations against various scenarios like owned vs borrowed data.

We are currently fixing some of this retroactively in `rust-secp`: https://github.com/rust-bitcoin/rust-secp256k1/pull/270